src/command_modules/azure-cli-eventgrid/azure/cli/command_modules/eventgrid/_params.py

```diff
@@ -27,15 +29,33 @@
     nargs='+'
 )
 
+odata_query_type = CLIArgumentType(
+    help="The OData query used for filtering the list results. Filtering is currently allowed on the Name property only. The supported operations include: CONTAINS, eq (for equal), ne (for not equal), AND, OR and NOT.",
+    options_list=['--odata-query']
+)
+
+
+domain_name_type = CLIArgumentType(
+    help='Name of the domain.',
+    arg_type=name_type,
+    options_list=['--domain-name'],
+    completer=get_resource_name_completion_list('Microsoft.EventGrid/domains'))
 
-def load_arguments(self, _):
+domain_topic_name_type = CLIArgumentType(
+    help='Name of the domain topic.',
+    arg_type=name_type,
+    options_list=['--domain-topic-name'],
+    completer=get_resource_name_completion_list('Microsoft.EventGrid/domains/topic'))
```

Comment:
Consider if we really need to have a completer for domain topics (since there could be 100,000 + of them). Not sure how this is implemented under the covers. Also, if we need a completer, I guess it needs to be topics instead of topic.

Reply:
Agreed. Trying it in shell.azure.com for domains, it seems it waits for all results to be shown before displaying them to user. Showing that for 100K domain topic is challenging. Removing it due to perf concern in enumerating the domain topics in more than 100K exists